### ISSUE TITLE
Check for expired resume token errors after polling a LRO more gracefully

### ISSUE THREAD

**Issue by gosua** [Azure/azure-sdk-for-go#19330]
# Problem
When I poll for a long running operation with an expired (older than ~20 minutes) resume token like so
```
poller, err := privateEndpointsClient.BeginCreateOrUpdate(ctx, "", "", armnetwork.PrivateEndpoint{}, &armnetwork.PrivateEndpointsClientBeginCreateOrUpdateOptions{
  ResumeToken: resumeToken // this resume token got created more than 20 minutes ago 
})
```
The SDK will return an error https://github.com/Azure/azure-sdk-for-go/blob/b386adc15c2e60e3812956f68d244b608330c479/sdk/azcore/internal/pollers/op/op.go#L98 but swallows the HTTP response https://github.com/Azure/azure-sdk-for-go/blob/b386adc15c2e60e3812956f68d244b608330c479/sdk/azcore/internal/pollers/op/op.go#L105

The only option I have in order to handle expired token errors is to check for the actual error string like so

```
if _, err := poller.Poll(ctx); err != nil {
  if err.Error() == "the response did not contain a status" { // azure-sdk swallows http response during polling. thus, need to handle invalid token error case based on error string
    // some recovering logic
  }
}
```

Executing the recovering logic now based on this error handling is super brittle since I am relying on a string value which can change quiet easily. 

# Feature Request
I wish for a way to inspect the HTTP response myself when the polling fails. Something like this _could_ do the job where the response of the poller https://github.com/Azure/azure-sdk-for-go/blob/b386adc15c2e60e3812956f68d244b608330c479/sdk/azcore/runtime/poller.go#L186 is exposed: 

```
if _, err := poller.Poll(ctx); err != nil {
  if poller.Resp.StatusCode == http.StatusNotFound { // Azure API returns 404 in case I poll for a LRO with an expired resume token
    ...
  }
}
```

More specific errors than `errors.New("...")` _might_ also be an option.

**Comment by jhendrixMSFT** [comment]
Agreed this needs improvement.  While we sort this out, you should be able to use `[runtime.WithCaptureResponse]`(https://pkg.go.dev/github.com/Azure/azure-sdk-for-go/sdk/azcore@v1.1.4/runtime#WithCaptureResponse) to get the raw HTTP response (not very elegant for this case but should get you unblocked).

**Comment by jhendrixMSFT** [comment]
The fix will be in `azcore@v1.2.0` which should be out in early November.
